### ISSUE TITLE
SUGGESTION: Reconsider or make the recent changes to menu & power button optional (Especially for those not using Onion Launcher)

### ISSUE THREAD

**Issue by Julian-MJK** [jimgraygit/Onion#134]
Edit: Elevating what I commented below: 
> **BIG NOTE: Especially for people who don't use Onion Launcher, I hadn't considered that**
> If someone uses Onion Launcher, accidentally pressing Menu is no big deal, you just press it again to go back, but for the people who haven't enabled Onion Launcher, the Menu button is much more destructive, then it would feel more right if you had to press menu+start to exit back to Miyoo. 

Luvmaya in #133 and Wapp92 in #135 have had the same issues as me with this change.

After upgrading my Onion from V3 to the latest, I find that the feeling of the console in general has gotten a lot more volatile, because of how easy it is to do "destructive" actions.

I always find myself accidentally quitting the game I'm inside when I'm just trying to change brightness, open the retroarch menu, or quickload/save. This happens because if you press the second button (up/down) just a tiny bit _before_ you hit the menu button, the Miyoo doesn't recognize that you pressed both of them, and just thinks you hit the menu button, and thus quits. 
I can get used to it, but this is likely an annoyance more people are experiencing as well. _I'll open a separate issue for this because that should probably be considered a bug._ 

The new power functionality is really awesome, but I think it might feel better if you have hold it down, like the stock Miyoo and other applications, because it is such a destructive action. But my primary issue is the menu button no longer being purely a modifier, I can see that instant power button is a decision that is integral to the new Onion Launcher experience.

When destructive actions are kept more than a millisecond away, you can't just instantly do them and instead have to e.g. hold the power button or press menu+start, the system feels "sturdier" somehow, less volatile, less easily changing, and thus feels more like a proper console, which rarely have such instant exit/shutdown methods.

**The new menu button also adds a new barrier to using it, before you could just tap or press it without any action happening, now you have to be certain you know what you want to do, before hitting it, because if you've pressed it down you _have_ to do an action, otherwise you exit the game. It means you hesitate, the menu button feels a tiny bit more like a mine that can go off if you do something wrong, throwing you out of the game, than the purely modifier button it used to be.** That's the core of my issue with these changes.

The new changes go against popular UX norms, so I think there are good reasons most consoles and systems and interfaces put up additional barriers between the user and destructive actions like these, like even Miyoo do in their stock firmware with the shutdown button, where you have to hold it down and then there's even a confirmation prompt, either of those would be great.

**Comment by Julian-MJK** [comment]
Luvmaya in issue #133 just brought up this exact issue, so that confirms I'm not the only one thinking this:
> Thanks, I understand but it seems nonsence for me. It is too easy to press menu button. I just press it by my mistake but... quit game?, tooo easy to quit game.. hmm anyway, I understood, Thanks

**Comment by doubled1** [comment]
Just my 2c

By default, both shutting down and exiting the game will take a save state.
It's not as destructive as it'd seem.  I prefer the fast response myself.

I've also not managed to hit the buttons out of order and get an unintended response.

**Comment by Julian-MJK** [comment]
After one of the devs sent a patch in the discord fixing the button issue, the system overall feels a lot less volatile than I experienced it when that was happening. So I'm not sure where I stand on this after that patch, the new UX is really quite good. But it's worth considering anyways

**Comment by Julian-MJK** [comment]
Changed the title as I think the new UX is splendid once the issue of hitting the buttons together was patched by a dev, which he said will be included in the next release of Onion.

But having a settings application to easily change things like the brightness shortcut, instead of having to hit START+SELECT+ A+B or whatever it was to change the brightness control from STARTSELECT+R1/R2 to MENU+UP/OWN, it would be nice with a separate settings application (assuming the stock settings app can't be changed?) that lets you do that, so that it's a discoverable option. And it's possible to do that in a settings applications, then other configurations could also be added, like requiring a long-press of the power button, or using the previous menu shortcut (menu+start) to make the menu button "non-volatile" like say keyboard modifiers are.

Not closing the issue myself in case other people want to chime in or something.

**Comment by Julian-MJK** [comment]
**BIG NOTE: Especially for people who don't use Onion Launcher, I hadn't considered that**
If someone uses Onion Launcher, accidentally pressing Menu is no big deal, you just press it again to go back, but for the people who haven't enabled Onion Launcher, the Menu button is much more destructive, then it would feel more right if you had to press menu+start to exit back to Miyoo. 

_I've even noticed this when quitting OnionLauncher myself, the Menu button becomes more dangerous._

**Comment by mvergaraq** [comment]
@Julian-MJK There is actually a Confirm quit setting in Retroarch (Under Hotkeys settings) that requires to press the button twice to quit, but it is not working.

I also don't have Onion Launcher activated and the menu button keeps feeling way too destructive.

Posible fixes:
> 1. Two menu presses to quit (ideally show a message "Press twice to quit" if user presses only once)
> 2. Press menu once and then a confirm dialog asks to quit the game (feels slower in terms of UX)
> 3. Menu+Start to exit back (Personally I don't like this option as there are already many combinations, and exiting a game shouldn't need it)

**Comment by mutcianm** [comment]
Not using the Onion Launcher as well.
In previous build I was able to unmap certain default hotkeys manually via retroarch settings, but unfortunately this is no longer an option for the "Exit game" hotkey.

IMO, the ability to control brightness and other quick access functions are great but it would be even better to still let the user have a choice whether to enable them or not.

**Comment by Sichroteph** [comment]
This button behavior has been made optional, check the wiki to disable it. In addition I also added Egg's custom keymon for a stock experience while keeping the latest features. Feel free to continue this debate in the discussion tab.
